### DISCUSSION TITLE
Efficient Pattern/Document Matching?

### DISCUSSION THREAD

**Discussion by sandbox-apps** [prisma/prisma#8611] [Q&A] [answered]
Hello guys, I have a query from Prisma and I logged it.
This is the output

**Find Many Groups**
```sql
SELECT 
"public"."Group"."id", 
"public"."Group"."name", 
"public"."Group"."shortname", 
"public"."Group"."description", 
"public"."Group"."coverUrl", 
"public"."Group"."photoUrl", 
"public"."Group"."createdAt", 
"public"."Group"."updatedAt" 
FROM "public"."Group" 
WHERE 1=1 
ORDER BY "public"."Group"."createdAt" DESC, "public"."Group"."id" DESC 
LIMIT $1 OFFSET $2
```

**From Find Many Groups with include members:true**
```sql
SELECT 
"public"."GroupMember"."groupId", 
"public"."GroupMember"."id", 
"public"."GroupMember"."role", 
"public"."GroupMember"."level", 
"public"."GroupMember"."experience", 
"public"."GroupMember"."reputation", 
"public"."GroupMember"."mutedUntil", 
"public"."GroupMember"."guardedUntil", 
"public"."GroupMember"."createdAt", 
"public"."GroupMember"."updatedAt" 
FROM "public"."GroupMember" 
WHERE "public"."GroupMember"."groupId" 
IN ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14,$15,$16,$17,$18,$19,$20,$21,$22,$23,$24,$25,$26,$27,$28,$29,$30,$31,$32) 
OFFSET $33
```

Just want to ask how do you filter and map the member response into their respective owner/parents which in this case the group. So when it returned from the API it is hierchically and relationally organized.

**Comment by ryands17** [comment, marked as answer]
@sandbox-apps 👋 

This is done in the [query engine](https://github.com/prisma/prisma-engines) written in Rust. The results are fetched separately as in the SQL queries above and then mapped as per the relation defined in `schema.prisma`.

**Reply by sandbox-apps** [reply]
Is both sql query sent to the postgres at the same time? 1 roundtrip?
Or it waits for the first one then send the 2nd one? 2 roundtrip?

@ryands17

**Reply by ryands17** [reply]
It waits for the first and then sends the second as they are two separate queries, so two roundtrips.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
